src/Helper/Normalizer.php

```diff
@@ -106,13 +105,20 @@ public static function renderLabel(
     ): string {
         $html = '';
 
+        $tagName = self::iconTagName($iconAttributes);
+
+        unset($iconAttributes['tag']);
+
         if ($iconClass !== '') {
             Html::addCssClass($iconAttributes, $iconClass);
         }
 
         if ($icon !== '' || $iconAttributes !== [] || $iconClass !== '') {
-            $i = I::tag()->attributes($iconAttributes)->content($icon);
-            $html = Span::tag()->attributes($iconContainerAttributes)->content($i)->encode(false)->render();
+            $html = Html::tag($tagName)->attributes($iconAttributes)->content($icon)->render();
+
+            if ($tagName === 'i') {
```

Comment:
Wrapping usage that depends from tag name is very unexpected.

This package contains common framework-agnostic widgets, so result should be defined clearly. Use container or not, should be defined clearly and not depend from tag name.

Reply:
The change is about the handling of icons, which are usually wrapped in a container, while SVGs, for example, are not wrapped, just like an image.